jena-arq/src/main/java/org/apache/jena/riot/lang/BlankNodeAllocatorHash.java

```diff
@@ -90,6 +80,12 @@ protected UUID freshSeed() {
     @Override
     public void reset() {
         UUID seed = this.freshSeed();
+        long mostSigBits = seed.getMostSignificantBits();
+        long leastSigBits = seed.getLeastSignificantBits();
+        // Stamp on version and variant. Makes it an illegal UUID (unless all the bits are zero!)
```

Comment:
I assume this is needed because otherwise the version and variant segments of the UUID would always be the same leading to some predictability in the hashing?

Reply:
TBH it is not needed, it's being a little tidy saying "this is not being used as a UUID". We are using a UUID but only as convenient way to get a quality random number so large it is unique for practical purposes. Version and variant are together 6 bits (fixed), leaving 122 bits of input. 122 bits is a very large number. IIRC (e.g. 3Store paper) a few bits more that 64 is enough. 58 bits more is a lot more.

> The “Birthday Paradox”1 gives the probability of an accidental hash collision
> occurring anywhere in a knowledge base of 10^9 distinct nodes to be around
> 1 : 10^−10 when using an effective 63 bit hash.

(more details in the paper)

The requirements are uniqueness for any bnodes that are ever going to meet (assumed "global"), speed and not requiring keeping the seen labels over the life of a parser run. Prior to using the hash allocator, Jena kept labels it had seen and very occasionally a user would run out of heap due to the footprint.

Reply:
Sure I'm well aware of the birthday paradox and the parsing concerns and overheads involved, just wanted to clarify the intent of the code